apps/imaginary.sh

```diff
@@ -0,0 +1,153 @@
+# GNU General Public License v3.0
+# https://github.com/nextcloud/vm/blob/master/LICENSE
+
+true
+SCRIPT_NAME="Imaginary Docker"
+SCRIPT_EXPLAINER="This script will install Imaginary which is a replacement for the less secure Imagick.
+It can speedup the loading of previews in Nextcloud a lot."
+# shellcheck source=lib.sh
+source /var/scripts/fetch_lib.sh
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+# Check recources
+ram_check 4
+cpu_check 4
+
+# Compatible with NC24 and above
+lowest_compatible_nc 26
+
+# Check if Imaginary is already installed
+if ! does_this_docker_exist nextcloud/aio-imaginary
+then
+    # Ask for installing
+    install_popup "$SCRIPT_NAME"
+else
+    # Ask for removal or reinstallation
+    reinstall_remove_menu "$SCRIPT_NAME"
+    # Removal
+    if yesno_box_no "Do you want to remove the Imaginary docker container and settings?"
+    then
+        if docker-compose_down "$SCRIPTS"/imaginary-docker/docker-compose.yml
+        then
+            countdown "Waiting for the Docker image to be destroyed" "5"
+            nextcloud_occ config:system:delete enabledPreviewProviders
+            nextcloud_occ config:system:delete preview_imaginary_url
+            rm -rf "$SCRIPTS"/imaginary-docker
+            docker system prune -a -f
+        fi
+    fi
+    # Remove everything that's related to previewgenerator - it's now legacy
+    nextcloud_occ app:remove previewgenerator
+    # reset the preview formats
+    nextcloud_occ_no_check config:system:delete "enabledPreviewProviders"
+    nextcloud_occ config:system:delete preview_max_x
+    nextcloud_occ config:system:delete preview_max_y
+    nextcloud_occ config:system:delete jpeg_quality
+    nextcloud_occ config:system:delete preview_max_memory
+    nextcloud_occ config:system:delete enable_previews
+    # reset the cronjob
+    crontab -u www-data -l | grep -v 'preview:pre-generate'  | crontab -u www-data -
+    # Remove apps
+    APPS=(php-imagick libmagickcore-6.q16-3-extra imagemagick-6.q16-extra)
+    for app in "${APPS[@]}"
+    do
+        if is_this_installed "$app"
+        then
+            apt-get purge "$app" -y
+        fi
+    done
+    if is_this_installed ffmpeg && ! is_app_installed integration_whiteboard
+    then
+        apt-get purge ffmpeg -y
+    fi
+    apt-get autoremove -y
+    if yesno_box_yes "Do you want to remove all previews that were generated until now?
+This will most likely clear a lot of space! Also, pre-generated previews are not needed anymore once Imaginary are installed."
+    then
+        countdown "Removing the preview folder. This can take a while..." "5"
+        rm -rfv "$NCDATA"/appdata_*/preview
+        print_text_in_color "$ICyan" "Scanning Nextclouds appdata directory after removing all previews. \
+This can take a while..."
+        nextcloud_occ files:scan-app-data -vvv
```

Comment:
This should be enough:

```suggestion
        nextcloud_occ files:scan-app-data preview -vvv
```

Reply:
Thanks